### DISCUSSION TITLE
Regenerate Util giving error... TypeError: _dna.split is not a function

### DISCUSSION THREAD

**Discussion by skyaam** [nftchef/art-engine#314] [Help + Q&A] [answered]
Please Help. All files are present in their respective original location. Whenever I try to regenerate a single image using the command "node utils/regenerate.js -d 1" it gives me the following error...

skyaam-MacBook-Pro Art_Engine % node utils/regenerate.js -d 1
/Users/skyaam/Art_Engine/src/config.js
Regemerating #1
Adding the following to incompatible list Orange Yellow
Adding the following to incompatible list Purple Blue
/Users/skyaam/Art_Engine/src/main.js:411
const filteredDNA = _dna.split(DNA_DELIMITER).filter((element) => {
^

TypeError: _dna.split is not a function
at filterDNAOptions (/Users/skyaam/Art_Engine/src/main.js:411:28)
at isDnaUnique (/Users/skyaam/Art_Engine/src/main.js:453:20)
at createItem (/Users/skyaam/Art_Engine/utils/regenerate.js:64:7)
at regenerateItem (/Users/skyaam/Art_Engine/utils/regenerate.js:122:35)
at Command. (/Users/skyaam/Art_Engine/utils/regenerate.js:180:5)
at Command.listener [as _actionHandler] (/Users/skyaam/Art_Engine/node_modules/commander/lib/command.js:488:17)
at /Users/skyaam/Art_Engine/node_modules/commander/lib/command.js:1227:65
at Command._chainOrCall (/Users/skyaam/Art_Engine/node_modules/commander/lib/command.js:1144:12)
at Command._parseCommand (/Users/skyaam/Art_Engine/node_modules/commander/lib/command.js:1227:27)
at Command.parse (/Users/skyaam/Art_Engine/node_modules/commander/lib/command.js:897:10)

**Comment by link-me** [comment, marked as answer]
Add .toString() method to all  _dna.split in src/main.js file

**Reply by skyaam** [reply]
Thanks @link-me for the suggestion. I tried it but I'm still getting the same error.

**Reply by link-me** [reply]
386: const dna = _dna.toString().split(DNA_DELiMITER);
435 const filtredDNA = dna.toString().split(DNA_DELiMITER).filter(element) => {

Try again, this working for me with same error

**Reply by skyaam** [reply]
That did the trick. Thank you @link-me.

**Comment by nftchef** [comment]
@skyaam , are you still having issues? 

what is the value for number 1 in your _dna.json file?

**Reply by skyaam** [reply]
@nftchef, I'm still having issues after adding .toString() to two _dna.split instances in src/main.js file as follows:

Line 362:    const dna = _dna.split(DNA_DELIMITER).toString();
Line 411:    const filteredDNA = _dna.split(DNA_DELIMITER).toString().filter((element) => {

And the value of no. 1 in the _dna.json file is:

"0.4:Grey#600.png*1.7:Pink#600.png*2.68:Bomber Jacket#69.png*3.3:Baseball#127.png*4.7:Fire#120.png*5.58:Stripes#105.png*6.0:Mask#250.png*7.14:Synthetic#420.png*8.10:Bam#670.png"

**Reply by LolaMcBurns** [reply]
Hey there! Still having issues after adding .toString() thingy as well. It works just fine regenerating selected editions with new ones actually. I keep hitting duplicates though (so something's up with DNA uniqueness check maybe?). Been trying to figure this out on my own for a few days now... ehm...no progress so far 😅 Would really appreciate it if some of you Wise Wizards could help find a solution because this util sounds absolutely amazing 🧙‍♂️👨‍💻 Thanks!
